scanner.go

```diff
@@ -137,6 +140,8 @@ func newScanner(c RPCClient, rpc *hrpc.Scan) *scanner {
 		startRow:           rpc.StartRow(),
 		curRegionScannerID: noScannerID,
 		scanMetrics:        sm,
+		// value is lease timeout / 2
+		renewInterval: 10 * time.Second,
```

Comment:
I dont think we should hardcode this since gohbase is open source and different users might have different lease timeout set. The problem is that the lease timeout is an hbase flag that you configure when starting up hbase so we don't have access to that number here AFAIK.

One way that I can think of is adding a flag `renewInterval` to the hrpc.Scan struct to set the time interval there and pass it down to here. Or we can change the hrpc.Scan.renew flag to be a pointer to `time.Duration` and only use it if it's not nil. @aaronbee lmk if you have other ideas or if I am wrong

Reply:
Implemented this with a slight change. Since converting to a protobuf is used internally within `SendRPC`, I split apart the `renew` boolean into 2, `renewInterval` which tracks the interval at which the scanner is renewed, and `renewalScan` which indicates whether this Scan request is a renewal request. These are both Scan options which can be set. Then in `ToProto` for `Scan` we check if `renewalScan` is true and if it is, then we set the `Renew` field in the protobuf messages accordingly. Let me know if this looks fine!